src/framework/mlt_factory.c

```diff
@@ -354,13 +324,20 @@ mlt_producer mlt_factory_producer( mlt_profile profile, const char *service, con
 		service = mlt_environment( "MLT_PRODUCER" );
 
 	// Offer the application the chance to 'create'
-	mlt_events_fire( event_object, "producer-create-request", service, resource, &obj, NULL );
+	mlt_factory_event_data data = {
```

Comment:
I do not  feel too strongly about these comments. But if you accept my suggestion to pass event_data by value, then these lines:

```
	mlt_factory_event_data data = {
	    .name = service,
	    .input = resource,
	    .service = &obj
	};
	mlt_event_data event_data;
	mlt_event_data_from_object(&event_data, &data);
	mlt_events_fire( event_object, "producer-create-request", &event_data );
```

Collapse to:

```
	mlt_factory_event_data data = {
	    .name = service,
	    .input = resource,
	    .service = &obj
	};
	mlt_events_fire( event_object, "producer-create-request", mlt_event_data_from_object(data) );
```

Reply:
I made all of these changes, and it is working. Overall I think it is cleaner, less pointer checks, Mlt::EventData no longer needs an `is_valid()`. Still reviewing all my changes before pushing it.